### PR TITLE
Remove Travis and fix CI in GitHub Actions

### PR DESCRIPTION
**Pull request by claudiosanches**
It started smaller, but I found some issues, so listing all changes here:

- Removed Travis CI integration.
- Fixed "Run code sniff on PR" workflow, removed unnecessary MySQL dependency, fixed `tests/bin/phpcs.sh` script to run with GitHub Actions and implemented `cs2pr` (there's examples of `cs2pr` output in this PR).
- Fixed "Run code coverage on PR" workflow, it wasn't sending any data to Codecov. You can check that's working with this last commit from this branch: https://codecov.io/gh/woocommerce/woocommerce/commit/4e893022a0f50da56b2376902ff72bbc3188b97d/
- Added new "Run CI" workflow, it runs in `trunk` and `release/**` branches, it makes easy for us to check status in those branches after merge commits and other changes, especially nice to check before releasing WooCommerce. And it's nice that it generates a new badge to replace our old Travis badge.

Notes about `cs2pr`: I triggered some errors in `includes/admin/settings/class-wc-settings-integrations.php` by just fixing one coding standard issue and leaving 3 others behind, I hope this test makes easy for reviewing this PR. Please let me know if you would like me to fix all coding standards issues before merging, so we have CI passing properly for this PR.

### CODE REVIEW

**Comment by Konamiman** [comment]
So if I understood correctly the few code sniffer violations are there on purpose to prove that `cs2pr` is working as expected? In that case I'd say that yes, it is, so I guess these can be fixed before merging.

Also I'm not sure if there anything else to do to test this PR, other than seeing that all the checks for the PR itself are passing 🙂

**Comment by claudiosanches** [comment]
@Konamiman yes, those are just to show that's working. I'm fixing those errors now. Thanks.

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 <a href="https://packagist.org/packages/woocommerce/woocommerce"><img src="https://poser.pugx.org/woocommerce/woocommerce/v/stable" alt="Latest Stable Version"></a>
 <img src="https://img.shields.io/wordpress/plugin/dt/woocommerce.svg" alt="WordPress.org downloads">
 <img src="https://img.shields.io/wordpress/plugin/r/woocommerce.svg" alt="WordPress.org rating">
-<a href="https://travis-ci.com/woocommerce/woocommerce"><img src="https://travis-ci.com/woocommerce/woocommerce.svg?branch=trunk" alt="Build Status"></a>
+<a href="https://github.com/woocommerce/woocommerce/actions/workflows/ci.yml"><img src="https://github.com/woocommerce/woocommerce/actions/workflows/ci.yml/badge.svg?branch=trunk" alt="Build Status"></a>
```
Comment (Konamiman) [inline README.md:8]: This looks weird. There's no `/actions` directory whatsoever (maybe you meant `/.github/workflows`?), there's no `badge.svg` file at all in the code, and I don't think that `ci.yml/badge.svg` would be the right path for it. Also, is it right to label a link to a static file as "Build status"?

image

**Review by Konamiman — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (claudiosanches) [reply README.md:8]: @Konamiman this will be only available once this PR gets merged into trunk, that's why is missing this badge.

Comment (claudiosanches) [reply README.md:8]: Here an example of our nightly builds:

https://github.com/woocommerce/woocommerce/actions/workflows/nightly-builds.yml
https://github.com/woocommerce/woocommerce/actions/workflows/nightly-builds.yml/badge.svg?branch=trunk


Build Status

Comment (Konamiman) [reply README.md:8]: Ok, got it, thanks

**Review by Konamiman — APPROVED** [review]

**Pull request merged**